zerver/actions/message_send.py

```diff
@@ -217,12 +225,22 @@ def get_recipient_info(
         # of this function for different message types.
         assert stream_topic is not None
 
+        topic_participant_user_ids: Set[int] = set()
+        if possible_topic_wildcard_mention:
+            # A topic participant is anyone who either sent or reacted to messages in the topic.
+            # It is expensive to call `participants_for_topic` if the topic has a large number
+            # of messages. But it is fine to call it here, as this gets called only if the message
+            # has @topic mentions.
```

Comment:
Most correctly, we only call this when there's syntax that might be an `@topic` mention, without having confirmed the syntax isn't, say, in a code block.

Reply:
Updated comment:
```python
  # A topic participant is anyone who either sent or reacted to messages in the topic.
  # It is expensive to call `participants_for_topic` if the topic has a large number
  # of messages. But it is fine to call it here, as this gets called only if the message
  # has syntax that might be a @topic mention without having confirmed the syntax isn't, say,
  # in a code block.
```